lib/cadet/assessments/assessments.ex

```diff
@@ -844,6 +862,114 @@ defmodule Cadet.Assessments do
     end
   end
 
+  # This function changes the is_grading_published field of the submission to false
+  # and sends a notification to the student
+  def unpublish_grading(submission_id, cr = %CourseRegistration{id: course_reg_id, role: role})
+      when is_ecto_id(submission_id) do
+    submission =
+      Submission
+      |> join(:inner, [s], a in assoc(s, :assessment))
+      |> preload([_, a], assessment: a)
+      |> Repo.get(submission_id)
+
+    # allows staff to unpublish own assessment
+    bypass = role in @bypass_closed_roles and submission.student_id == course_reg_id
+
+    with {:submission_found?, true} <- {:submission_found?, is_map(submission)},
+         {:allowed_to_unpublish?, true} <-
+           {:allowed_to_unpublish?,
+            role == :admin or bypass or
+              Cadet.Accounts.Query.avenger_of?(cr, submission.student_id)} do
+      Multi.new()
+      |> Multi.run(
```

Comment:
I think there is no need to use multi and transaction here since there is only a single update. May II ask what was the reason for this?